Flaky TestIcebergGlueTableOperationsInsertFailure.testInsertFailureDoesNotCorruptTheTableMetadata

```
Error:  io.trino.plugin.iceberg.catalog.glue.TestIcebergGlueTableOperationsInsertFailure.testInsertFailureDoesNotCorruptTheTableMetadata -- Time elapsed: 3.525 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual throwable to be an instance of:
  org.apache.iceberg.exceptions.CommitStateUnknownException
but was:
  io.trino.testing.QueryFailedException: Test-simulated Glue timeout exception
Cannot determine whether the commit was successful or not, the underlying data files may or may not be needed. Manual intervention via the Remove Orphan Files Action can remove these files when a connection to the Catalog can be re-established if the commit was actually unsuccessful.
Please check to see whether or not your commit was successful before retrying this commit. Retrying an already successful operation will result in duplicate records or unintentional modifications.
At this time no files will be deleted including possibly unused manifest lists.
	at io.trino.testing.DirectTrinoClient.toMaterializedRows(DirectTrinoClient.java:159)
	at io.trino.testing.DirectTrinoClient.execute(DirectTrinoClient.java:135)
	at io.trino.testing.DirectTrinoClient.execute(DirectTrinoClient.java:84)
	...(55 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
	at io.trino.plugin.iceberg.catalog.glue.TestIcebergGlueTableOperationsInsertFailure.testInsertFailureDoesNotCorruptTheTableMetadata(TestIcebergGlueTableOperationsInsertFailure.java:127)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```
https://github.com/trinodb/trino/actions/runs/7658421188/job/20871459013

--- a/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueTableOperationsInsertFailure.java
+++ b/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueTableOperationsInsertFailure.java
@@ -17,14 +17,15 @@ import com.amazonaws.services.glue.AWSGlueAsync;
 import com.google.common.collect.ImmutableMap;
 import io.airlift.log.Logger;
 import io.trino.Session;
+import io.trino.execution.Failure;
 import io.trino.plugin.hive.metastore.Database;
 import io.trino.plugin.hive.metastore.glue.GlueHiveMetastore;
 import io.trino.plugin.iceberg.TestingIcebergPlugin;
 import io.trino.spi.security.PrincipalType;
 import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryFailedException;
 import io.trino.testing.QueryRunner;
 import io.trino.testing.StandaloneQueryRunner;
-import org.apache.iceberg.exceptions.CommitStateUnknownException;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
@@ -40,6 +41,7 @@ import static io.trino.plugin.hive.metastore.glue.TestingGlueHiveMetastore.creat
 import static io.trino.testing.TestingNames.randomNameSuffix;
 import static io.trino.testing.TestingSession.testSessionBuilder;
 import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
 
@@ -124,8 +126,13 @@ public class TestIcebergGlueTableOperationsInsertFailure
 
         getQueryRunner().execute(format("CREATE TABLE %s (a_varchar) AS VALUES ('Trino')", tableName));
         assertThatThrownBy(() -> getQueryRunner().execute("INSERT INTO " + tableName + " VALUES 'rocks'"))
-                .isInstanceOf(CommitStateUnknownException.class)
-                .hasMessageContaining("Test-simulated Glue timeout exception");
+                .satisfies(throwable -> {
+                    assertThat(throwable).isInstanceOf(QueryFailedException.class);
+                    assertThat(throwable.getCause()).isInstanceOf(Failure.class);
+                    Failure failure = (Failure) throwable.getCause();
+                    assertThat(failure.getMessage()).contains("Test-simulated Glue timeout exception");
+                    assertThat(failure.getFailureInfo().getType()).isEqualTo("org.apache.iceberg.exceptions.CommitStateUnknownException");
+                });
         assertQuery("SELECT * FROM " + tableName, "VALUES 'Trino', 'rocks'");
     }
 }

